urla05239b71c_viniciusinnovatium_mumps2java_api_tgz-pJ8-com_disclinc_netmanager_script_test_ExecTesteNewJ8/src/mLibrary/mContext.java

```diff
@@ -0,0 +1,525 @@
+package mLibrary;
+
+import java.io.Writer;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import mSystem.mSystem;
+import br.com.innovatium.mumps2java.todo.REMOVE;
+import br.com.innovatium.mumps2java.todo.TODO;
+
+public class mContext {
+	// TODO remover
+	public int xecuteCount;
+	private mData mDataPublic;
+	private mData mDataGlobal;
+	private mData mDataLocal;
+	private int countNewOperator;
+	private Map<String, Method> methodMap;
+
+	private mRequest mReq;
+	private mSession mSes;
+	public mFnc Fnc;
+	public mCmd Cmd;
+	private mSystem system;
+	private Writer writer;
+
+	private Map<String, Class> stackedClasses = new HashMap<String, Class>(30);
+
+	public mContext(Writer writer) {
+		this();
+		this.writer = writer;
+	}
+
+	public mContext() {
+		this.mDataPublic = new mData();
+		this.mDataGlobal = new mData();
+		this.mDataLocal = new mData();
+
+		this.Fnc = new mFnc(this);
+		this.Cmd = new mCmd(this);
+		this.system = new mSystem(this);
+	}
+
+	public Writer getWriter() {
+		return writer;
+	}
+
+	public mData getmDataPublic() {
+		return mDataPublic;
+	}
+
+	public boolean hasPublicVariables() {
+		return !mDataPublic.isEmpty();
+	}
+
+	public mData getmDataGlobal() {
+		return mDataGlobal;
+	}
+
+	public mData getmDataLocal() {
+		return mDataLocal;
+	}
+
+	public mSystem getSystem() {
+		return system;
+	}
+
+	public String dump() {
+		return mDataPublic.dump();
+	}
+
+	public Object dispatch(mClass objClass, String methodName,
+			Object... parameters) {
+		Method m = getMethod(methodName);
+		Object result = null;
+		Object obj = null;
+		int countOld = countNewOperator;
+		countNewOperator = 0;
+		try {
+			if (objClass != null) {
+				obj = objClass;
+			} else if (!Modifier.isStatic(m.getModifiers())) {
+				obj = m.getDeclaringClass().newInstance();
+				if (obj instanceof mClass) {
+					((mClass) obj).setContext(this);
+				}
+			}
+			if (m.getParameterTypes() != null
+					&& m.getParameterTypes().length > 0
+					&& m.getParameterTypes()[0].isArray()) {
+				parameters = new Object[] { parameters };
+			}
+			if (m.getReturnType().equals(Void.TYPE)) {
+				m.invoke(obj, parameters);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)